cerebrium/scaling/graceful-termination.mdx

```diff
@@ -0,0 +1,201 @@
+---
+title: "Preemption and Graceful Termination"
+description: "Prevent 502 errors by handling SIGTERM signals during instance shutdown"
+---
+
+## Graceful Termination
+
+Cerebrium is a shared-compute, multi-tenant environment. To scale out, optimize compute packing, and perform updates, Cerebrium dynamically adjusts its capacity—shutting down nodes and bringing up new ones throughout the day. As this happens, user workloads are shifted to new nodes. In addition, your application has its own metric-based autoscaling criteria that dictate when instances should scale or remain active, as well as handle instance shifting during new app deployments. You may need to implement graceful shutdown to prevent requests from ending prematurely when app instances terminate, which commonly helps avoid 502 errors.
+
+## Understanding Instance Termination
+
+```mermaid
+stateDiagram-v2
+  direction LR
+
+  [*] --> Running
+
+  Running --> SIGTERM: scale down / redeploy
+  state "SIGTERM sent" as SIGTERM
+
+  SIGTERM --> ifGrace
+  state ifGrace <<choice>>
+
+  ifGrace --> Waiting: response_grace_period configured
+  ifGrace --> ForceKilled: no response_grace_period
+
+  state "Waiting (grace period)" as Waiting
+  Waiting --> GracefulShutdown: all requests complete
+  Waiting --> ForceKilled: grace period timeout
+
+  GracefulShutdown --> [*]: clean shutdown
+  ForceKilled --> [*]: 502 errors
+
+  note right of SIGTERM
+    Cortex: Automatic
+    Custom: Must implement
+  end note
+
+  note right of Waiting
+    Application waits for
+    active requests to complete
+  end note
+```
+
+For both application autoscaling and our own internal node scaling, we will send your application a SIGTERM signal, as a warning to the application that we are intending to shut down this instance. For Cortex applications, this is handled. On custom runtimes, should you wish to gracefully shut down, you will need to catch and handle this signal. Once at least responseGracePeriod has elapsed, we will send your application a SIGKILL signal, terminating the instance immediately.
+
+When Cerebrium needs to terminate an instance:
+
+1. Stops routing new requests to the instance
+2. Sends SIGTERM signal to your container
+3. Waits for `response_grace_period` seconds (if configured)
+4. Sends SIGKILL if the instance hasn't stopped
+
+Without `response_grace_period` configured, Cerebrium force-kills instances immediately, causing 502 errors.
```

Comment:
Its not clear that if if I don't handle the sigterm, then the default is the response grace period (we should have a default value here). Its also not clear that I handle the sigterm how that interacts with the response grace period

I am guessing this is for cortex then? maybe just be specific?